tutorials/shading/your_first_shader/your_first_canvasitem_shader.rst

```diff
@@ -0,0 +1,212 @@
+.. _doc_your_first_canvasitem_shader:
+
+Your first CanvasItem shader
+============================
+
+Introduction
+------------
+
+Shaders are special programs that execute on the GPU and are used for rendering 
+graphics. All modern rendering is done with shaders. For a more detailed description
+of what shaders are please see :ref:`What are shaders <doc_what_are_shaders>`.
+
+This tutorial will focus on the practical aspects of writing shader programs by walking
+readers through the process of writing a shader with both vertex and fragment functions.
+This tutorial targets absolute beginners to shaders.
+
+.. note:: If you have experience writing shaders and are just looking for 
+          an overview of how shaders work in Godot, see the :ref:`Shading Reference <toc-shading-reference>`.
+
+Setup
+-----
+
+:ref:`CanvasItem <doc_canvas_item_shader>` shaders are used to draw all 2D objects in Godot, 
+while :ref:`Spatial <doc_spatial_shader>` shaders are used to draw all 3D objects.
+
+In order to use a shader it must be attached inside a :ref:`Material <class_material>` 
+which must be attached to an object. To draw multiple objects with the same material, 
+the material must be attached to each object.
```

Comment:
I think it would be good to mention that Materials are Resources. Perhaps also link to https://docs.godotengine.org/en/latest/getting_started/step_by_step/resources.html